buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -2534,9 +2534,13 @@ class BeamModulePlugin implements Plugin<Project> {
         dependsOn setupTask
         dependsOn config.startJobServer
       }
-      mainTask.configure{dependsOn goTask}
-      cleanupTask.configure{mustRunAfter goTask}
-      config.cleanupJobServer.configure{mustRunAfter goTask}
+      // CrossLanguageValidatesRunnerTask is setup under python sdk but also runs tasks not involving
+      // python versions. set 'skipNonPythonTask' property to avoid duplicated run of these tasks.
+      if (!(project.hasProperty('skipNonPythonTask') && project.skipNonPythonTask == 'true')) {
+        mainTask.configure { dependsOn goTask }
+      }
+      cleanupTask.configure { mustRunAfter goTask }
+      config.cleanupJobServer.configure { mustRunAfter goTask }
```

Comment:
Do we need to special case these as well so that they run even if we're not depending on the goTask?

Reply:
Sorry, answered wrong the yes/no question. I mean run after dependencies are always configured not relevant to the dependency of validatesCrossLanguageRunnerGoUsingJava -> validatesCrossLanguageRunner, so I consider the answer is no

Reply:
Ah, I see - thanks!